core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java

```diff
@@ -42,6 +42,16 @@
 public class MatchQueryIT extends ESIntegTestCase {
     private static final String INDEX = "test";
 
+
+    /**
+     * Make sure we always have a single shard since we will be testing cutoff freq which
+     * is per-shard
+     */
+    @Override
+    protected int numberOfShards() {
+        return 1;
```

Comment:
I worry that this may end up hiding some cross-shard issues for other queries. Perhaps this should be controlled via the tests that need it (possibly using DFS if that works?)?

Since I'm not familiar with what Lucene's Graph query is even doing, I just wanted to put this out there.